File: APPManage/AM.Model/basket_check.cs
/**  版本信息模板在安装目录下，可自行修改。
* basket_check.cs
*
* 功 能： N/A
* 类 名： basket_check
*
* Ver    变更日期             负责人  变更内容
* ───────────────────────────────────
* V0.01  2016/8/12 17:55:15   N/A    初版
*
* Copyright (c) 2012 Maticsoft Corporation. All rights reserved.
*┌──────────────────────────────────┐
*│　此技术信息为本公司机密信息，未经本公司书面同意禁止向第三方披露．　│
*│　版权所有：动软卓越（北京）科技有限公司　　　　　　　　　　　　　　│
*└──────────────────────────────────┘
*/
using System;
namespace AM.Model
{
	/// <summary>
	/// basket_check:实体类(属性说明自动提取数据库字段的描述信息)
	/// </summary>
	[Serializable]
	public partial class basket_check
	{
		public basket_check()
		{}
		#region Model
		private string _guid;
		private string _userid;
		private DateTime? _createdate;
		private string _cpmc;
		private string _ggxh;
		private string _ypbh;
		private string _ypsl;
		private string _scrq;
		private string _jylx;
		private string _wtdw;
		private string _scdw;
		private string _jydd;
		private string _jysj;
		private string _jyyj;
		private string _jyjl;
		private string _bz;
		/// <summary>
		/// 
		/// </summary>
		public string Guid
		{
			set{ _guid=value;}
			get{return _guid;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string userid
		{
			set{ _userid=value;}
			get{return _userid;}
		}
		/// <summary>
		/// 
		/// </summary>
		public DateTime? CreateDate
		{
			set{ _createdate=value;}
			get{return _createdate;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string cpmc
		{
			set{ _cpmc=value;}
			get{return _cpmc;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ggxh
		{
			set{ _ggxh=value;}
			get{return _ggxh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ypbh
		{
			set{ _ypbh=value;}
			get{return _ypbh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ypsl
		{
			set{ _ypsl=value;}
			get{return _ypsl;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string scrq
		{
			set{ _scrq=value;}
			get{return _scrq;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string jylx
		{
			set{ _jylx=value;}
			get{return _jylx;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string wtdw
		{
			set{ _wtdw=value;}
			get{return _wtdw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string scdw
		{
			set{ _scdw=value;}
			get{return _scdw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string jydd
		{
			set{ _jydd=value;}
			get{return _jydd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string jysj
		{
			set{ _jysj=value;}
			get{return _jysj;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string jyyj
		{
			set{ _jyyj=value;}
			get{return _jyyj;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string jyjl
		{
			set{ _jyjl=value;}
			get{return _jyjl;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string bz
		{
			set{ _bz=value;}
			get{return _bz;}
		}
		#endregion Model

	}
}

File: AM/Scripts/appm/basket_check.js
function test() {
    var isok = false;
    if ($("#cpname").val() == "") {
        alert("产品名称不能为空");
        $("#cpname").focus();
    } else {
        if ($("#type").val() == "") {
            alert("规格型号不能为空");
            $("#type").focus();
        } else {
            if ($("#ypcode").val() == "") {
                alert("样品编号不能为空");
                $("#ypcode").focus();
            } else {
                if ($("#ypnumber").val() == "") {
                    alert("样品数量不能为空");
                    $("#ypnumber").focus();
                } else {
                    if ($("#scdate").val() == "") {
                        alert("生产日期不能为空");
                        $("#scdate").focus();
                    } else {
                        if ($("#checktype").val() == "") {
                            alert("检验类别不能为空");
                            $("#checktype").focus();
                        } else {
                            if ($("#wtcompany").val() == "") {
                                alert("委托单位不能为空");
                                $("#wtcompany").focus();
                            } else {
                                if ($("#sccompany").val() == "") {
                                    alert("生产单位不能为空");
                                    $("#sccompany").focus();
                                } else {
                                    if ($("#checkaddress").val() == "") {
                                        alert("检验地点不能为空");
                                        $("#checkaddress").focus();
                                    } else {
                                        if ($("#checktime").val() == "") {
                                            alert("检验时间不能为空");
                                            $("#checktime").focus();
                                        } else {
                                            if ($("#checkyj").val() == "") {
                                                alert("检验依据不能为空");
                                                $("#checkyj").focus();
                                            } else {
                                                if ($("#checkend").val() == "") {
                                                    alert("检验结论不能为空");
                                                    $("#checkend").focus();
                                                }
                                                else {
                                                    isok = true;
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
    return isok;
}

String.prototype.trim = function () {

    return this.replace(/(^\s*)|(\s*$)/g, '');
}

function submit() {
    var check = test();
    if (check) {
        var userid = $("#userid").val();
        var objArr = $("input[type=text]")
        //var checkitem = $("input[name=checkbox_name]:checked");
        //var radios = $("input[name=sex]:checked");
        //var xb = $(radios[0]).next("label").text();
        var textareas = $("textarea");
        //var pxd = "";
        //for (var i = 0; i < checkitem.length; i++) {

        //    pxd += $(checkitem[i]).parent("label").text().trim();
        //    if (i < checkitem.length) {
        //        pxd += ","
        //    }
        //}
        var json = {
            "basket_check": [{
                "Guid": "",
                "CreateDate": "",
                "userid": userid,
                "cpmc": $(objArr[0]).val(),
                "ggxh": $(objArr[1]).val(),
                "ypbh": $(objArr[2]).val(),
                "ypsl": $(objArr[3]).val(),
                "scrq": $(objArr[4]).val(),
                "jylx": $(objArr[5]).val(),
                "wtdw": $(objArr[6]).val(),
                "scdw": $(objArr[7]).val(),
                "jydd": $(objArr[8]).val(),
                "jysj": $(objArr[9]).val(),
                "jyyj": $(textareas[0]).val(),
                "jyjl": $(textareas[1]).val(),
                "bz": $(textareas[2]).val()
            }]
        };
        $.ajax({
            type: "post",
            url: "/AppM/addbasket_check",
            data: { obj: JSON.stringify(json) },
            success: function (data) {
                layer.msg("提交成功");
            }
        })
    }
}